editoast/src/views/scenario.rs

```diff
@@ -231,31 +284,43 @@ async fn patch(
     data: Json<ScenarioPatchForm>,
     path: Path<(i64, i64, i64)>,
     db_pool: Data<DbPool>,
-) -> Result<Json<ScenarioWithDetails>> {
+) -> Result<Json<ScenarioResponse>> {
     let (project_id, study_id, scenario_id) = path.into_inner();
 
-    // Check if project and study exist
-    let (project, study) = check_project_study(db_pool.clone(), project_id, study_id)
-        .await
-        .unwrap();
-
-    // Update a scenario
-    let scenario: Scenario = data.into_inner().into();
-    let scenario = match scenario.update(db_pool.clone(), scenario_id).await? {
-        Some(scenario) => scenario,
-        None => return Err(ScenarioError::NotFound { scenario_id }.into()),
-    };
-
-    // Update study last_modification field
-    let study = study.update_last_modified(db_pool.clone()).await?;
-    study.expect("Study should exist");
-
-    // Update project last_modification field
-    let project = project.update_last_modified(db_pool.clone()).await?;
-    project.expect("Project should exist");
-
-    let scenarios_with_details = scenario.with_details(db_pool).await?;
-    Ok(Json(scenarios_with_details))
+    let mut conn = db_pool.get().await?;
+    let mut project = Default::default();
+    let mut study = Default::default();
+
+    let scenarios_with_details = conn
+        .transaction::<_, InternalError, _>(|conn| {
+            async {
+                // Check if project and study exist
+                (project, study) = check_project_study_conn(conn, project_id, study_id)
+                    .await
+                    .unwrap();
+
+                // Update a scenario
+                let scenario: Scenario = data.into_inner().into();
+                let scenario = match scenario.update_conn(conn, scenario_id).await? {
+                    Some(scenario) => scenario,
+                    None => return Err(ScenarioError::NotFound { scenario_id }.into()),
+                };
+
+                // Update study last_modification field
+                let _study = study.clone().update_last_modified_conn(conn).await?;
+                study = _study.expect("Study should exist");
+
+                // Update project last_modification field
+                let _project = project.clone().update_last_modified_conn(conn).await?;
+                project = _project.expect("Project should exist");
+                scenario.with_details(db_pool).await
+            }
+            .scope_boxed()
+        })
+        .await?;
```

Comment:
You don't need mutability here. Rather than leaving the project and study uninitialized, you can just return them from the transaction.

```suggestion
    let (project, study, scenario) = conn
        .transaction::<_, InternalError, _>(|conn| {
            async {
                // Check if project and study exist
                let (project, study) = check_project_study_conn(conn, project_id, study_id)
                    .await
                    .unwrap();

                // Update the scenario
                let scenario: Scenario = data.into_inner().into();
                let scenario = match scenario.update_conn(conn, scenario_id).await? {
                    Some(scenario) => scenario,
                    None => return Err(ScenarioError::NotFound { scenario_id }.into()),
                };

                // Update study last_modification field
                let study = study.clone().update_last_modified_conn(conn).await?.expect("Study should exist");

                // Update project last_modification field
                let project = project.clone().update_last_modified_conn(conn).await?.expect("Project should exist");
                (project, study, scenario.with_details(db_pool).await)
            }
            .scope_boxed()
        })
        .await?;
```